### PR TITLE
Implement display for Balance and Coin

### PR DESCRIPTION
**Pull request by orkunkl**
Need this for a project and thought would be great to have it in std.
Not sure if formatting follows best practices.

### CODE REVIEW

**Comment by uint** [comment]
Looks good. I don't think there are best practices for `Display` formatting except "it's for user-facing output, so make it look good."

[`cosmwasm_std` already does this for `Coin` in a different format though.](https://github.com/CosmWasm/cosmwasm/blob/579dbfed2b337ce000abfaaae29d130d22393bac/packages/std/src/coins.rs#L22-L30) It'd probably be good to be consistent.

File: packages/cw0/src/balance.rs [hunk 1]
```diff
@@ -91,6 +91,15 @@ impl NativeBalance {
     }
 }
 
+impl fmt::Display for NativeBalance {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        for c in &self.0 {
+            write!(f, "denom: {}, amount: {}", c.denom, c.amount)?
```
Comment (orkunkl) [inline packages/cw0/src/balance.rs:97]:
```suggestion
            write!(f, "{}{}", c.denom, c.amount)?
```

**Review by uint — APPROVED** [review]
LGTM

**Review by ueco-jb — APPROVED** [review]
lgtm

**Pull request merged**